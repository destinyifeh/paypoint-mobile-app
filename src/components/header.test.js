'use strict';

import React from 'react';
import { Provider } from 'react-redux';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

import Header from './header';

configure({
  adapter: new Adapter()
});

const mockStore = configureStore([]);

describe('Header test', () => {
  let store;
  let component;
  let componentWithNavigator;
  let componentWithNavigateBackIcon;
  let componentWithoutVisibleNavigator;
  let componentWithVisibleNavigator;

  const hideNavigationMenu = jest.fn();
  const showNavigationMenu = jest.fn();

  beforeEach(() => {
    store = mockStore({
      navigation: {
        isNavigatorVisible: true
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );

    componentWithNavigator = renderer.create(
      <Provider store={store}>
        <Header 
          withNavigator
        />
      </Provider>
    );

    componentWithNavigateBackIcon = renderer.create(
      <Provider store={store}>
        <Header 
          withNavigateBackIcon
        />
      </Provider>
    );

    componentWithoutVisibleNavigator = shallow(
      <Provider store={store}>
        <Header 
          hideNavigationMenu={hideNavigationMenu}
          showNavigationMenu={showNavigationMenu}
          isNavigatorVisible={false}
          withNavigateBackIcon
        />
      </Provider>
    ).dive()

    componentWithVisibleNavigator = shallow(
      <Provider store={store}>
        <Header 
          hideNavigationMenu={hideNavigationMenu}
          showNavigationMenu={showNavigationMenu}
          isNavigatorVisible={true}
          withNavigateBackIcon
        />
      </Provider>
    ).dive()

  })
  
  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('with navigator remains the same', () => {
    expect(componentWithNavigator.toJSON()).toMatchSnapshot();
  });

  it('with navigate back remains the same', () => {
    expect(componentWithNavigateBackIcon).toMatchSnapshot();
  });

  it('with visible navigator remains the same', () => {
    expect(componentWithVisibleNavigator).toMatchSnapshot();
  });

  it('without visible navigator remains the same', () => {
    expect(componentWithoutVisibleNavigator).toMatchSnapshot();
  });

  it('without visible navigator onPress shows navigator', () => {
    componentWithoutVisibleNavigator.find(TouchableOpacity).first().props().onPress();
    expect(showNavigationMenu.mock.calls.length).toBe(1);
  });

  // it('with navigator remains the same', () => {
  //   const tree = renderer
  //     .create(<Header
  //       withNavigateBackIcon
  //     />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });git chce

})
