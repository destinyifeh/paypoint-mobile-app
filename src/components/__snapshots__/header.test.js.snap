// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "borderColor": "transparent",
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 20,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Header test with navigate back remains the same 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "borderColor": "transparent",
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 20,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="transparent"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#EE312A",
                  "fontSize": 36,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Header test with navigator remains the same 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "borderColor": "transparent",
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 20,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="transparent"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 36,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Header test with visible navigator remains the same 1`] = `ShallowWrapper {}`;

exports[`Header test without visible navigator remains the same 1`] = `ShallowWrapper {}`;
